Source/WebCore/svg/SVGElement.cpp

```diff
@@ -263,7 +263,7 @@ void SVGElement::reportAttributeParsingError(SVGParsingError error, const Qualif
 
 void SVGElement::removedFromAncestor(RemovalType removalType, ContainerNode& oldParentOfRemovedTree)
 {
-    if (removalType.disconnectedFromDocument)
+    if (removalType.disconnectedFromDocument && selfHasRelativeLengths())
         updateRelativeLengthsInformation(false, *this);
```

Comment:
This looks hard to understand. We are passing hasRelativeLengths = false to updateRelativeLengthsInformation() only when `selfHasRelativeLengths()` is true.

\Maybe we need to rename `hasRelativeLengths` to `needToUpdateRelativeLengths` or `needToRegisterForRelativeLengths` or something else. We do not necessarily pass the real value of `hasRelativeLengths` to `updateRelativeLengthsInformation()`.